File: AlumniDarli/app/Models/Event.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Event extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'title',
        'category',
        'date',
        'time',
        'location',
        'description',
        'image',
        'status',
        'status_admin',
        'status_pimpinan',
    ];

    public function user()
    {
        return $this->belongsTo(User::class, 'user_id', 'id_user');
    }

    public function participants()
    {
        return $this->belongsToMany(User::class, 'event_user', 'event_id', 'user_id')
                    ->withPivot('status')
                    ->withTimestamps();
    }
}

File: Darli/app/src/main/java/com/example/darli/data/model/Event.kt
package com.example.darli.data.model

data class Event(
    val id: Int,
    val title: String?,
    val category: String?,
    val date: String?,
    val raw_date: String?,
    val time: String?,
    val location: String?,
    val description: String?,
    val image: String?,
    val status: String?,
    val user_id: Int?,
    val creator_name: String?,
    val is_joined: Boolean?,
    val status_admin: String?,
    val status_pimpinan: String?
)

data class EventResponse(
    val response_code: Int,
    val content: List<Event>
)
